### PR TITLE
Site: Document support for custom Envoy JSON Fields

### PR DESCRIPTION
**Pull request by XanderStrike**
follow-up for https://github.com/projectcontour/contour/pull/3059

### CODE REVIEW

File: site/_guides/structured-logs.md [hunk 1]
```diff
@@ -60,7 +78,8 @@ json-fields:
   - "x_forwarded_for"
 ```
 
-[1]: https://godoc.org/github.com/projectcontour/contour/internal/envoy#JSONFields
-[2]: https://godoc.org/github.com/projectcontour/contour/internal/envoy#DefaultFields
-[3]: {{site.github.repository_url}}/issues/1507
-[4]: {{site.github.repository_url}}/blob/{{site.github.latest_release.tag_name}}/examples/contour/01-contour-config.yaml
+[1]: https://github.com/projectcontour/contour/blob/main/pkg/config/accesslog.go#L33-L45
```
Comment (youngnick) [inline site/_guides/structured-logs.md:81]: I'd prefer these to point to `godoc.org` with the `#ObjectName` on the end of the link - this seems a bit fragile, and would need updating whenever we change `accesslog.go`.

**Review by youngnick — CHANGES_REQUESTED** [review]
Mostly LGTM, but I'd really like the reference links to go to `godoc.org`, as I think that will be more stable than line numbers in Github.

File: site/_guides/structured-logs.md (hunk 1 above)
Comment (XanderStrike) [reply site/_guides/structured-logs.md:81]: Sorry, I should've explained this decision in the PR comment, totally slipped my mind.

As a result of some feedback on the previous PR where we implemented this feature, we stopped exporting JSONFields and didn't export envoy{Simple,Complex}Operators either, as they weren't required outside the package. As a result, they don't appear on what I believe to be the [relevant godoc page](https://godoc.org/github.com/projectcontour/contour/pkg/config#pkg-variables). I figured that the possibility of the link drifting off by a few lines over time was more desirable than no links, or the dead links that were there before.

If there's another page I can point to, or some godoc configuration that will allow it to expose these variables, I'm happy to make that change.

Comment (youngnick) [reply site/_guides/structured-logs.md:81]: Ah, right, that makes total sense. Line numbers will have to do then.

**Review by youngnick — APPROVED** [review]
LGTM

**Review by jpeach — APPROVED** [review]

**Pull request merged**